cbot_control/src/control_node.cpp

```diff
@@ -1,14 +1,20 @@
 #include "ros/ros.h"
 #include "cbot_ros_msgs/AHRS.h"
+#include "nav_msgs/Odometry.h"
 #include "cbot_ros_msgs/ControllerSettings.h"
 #include "cbot_ros_msgs/ThrusterControl.h"
 #include "cbot_ros_msgs/ControllerInputs.h"
 #include "cbot_control/controllers.hpp"
 
-double desired_heading, desired_thrust;
-double yaw, yaw_rate;
-double common_mode = 0, differential_mode = 0;
-int Controller_ON;
+#define PI 3.141592653
+#define deg2rad 0.0174532925
+#define rad2deg 57.2957795
+
+double desired_heading, desired_thrust, desired_pitch, desired_depth=1, desired_u=0;
+double yaw, yaw_rate, pitch, pitch_rate,roll=0, last_pitch, last_pitch_rate, Ts, depth,Vx,Vy,Vz;
+double common_mode_F = 0, differential_mode_F = 0;
+double common_mode_V = 0, differential_mode_V = 0;
+int Controller_ON, HeadingCtrlON, PitchCtrlON, VelocityCtrlON, DepthCtrlON, first_diff_pitch=0, flag=0;
```

Comment:
It will be better to create class when there are so many global vars. But I guess fine for now!